tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * @group editor
+ *
+ * @covers ::_disable_content_editor_for_navigation_post_type
+ */
+class Tests_Editor_DisableContentEditorForNavigationPostType extends WP_UnitTestCase {
+
+	const NAVIGATION_POST_TYPE     = 'wp_navigation';
+	const NON_NAVIGATION_POST_TYPE = 'wp_non_navigation';
+
+	public function tear_down() {
+		$this->enable_editor_support();
+		parent::tear_down();
+	}
+
+	/**
+	 * @ticket 56266
+	 */
+	public function test_it_doesnt_disable_content_editor_for_non_navigation_type_posts() {
+		$post = $this->create_non_navigation_post();
+		$this->assertTrue( $this->supports_block_editor() );
+
+		_disable_content_editor_for_navigation_post_type( $post );
+
+		$this->assertTrue( $this->supports_block_editor() );
+	}
+
+	/**
+	 * @ticket 56266
+	 */
+	public function test_it_disables_content_editor_for_navigation_type_posts() {
+		$post = $this->create_navigation_post();
+		$this->assertTrue( $this->supports_block_editor() );
+
+		_disable_content_editor_for_navigation_post_type( $post );
+
+		$this->assertFalse( $this->supports_block_editor() );
+	}
+
+	private function supports_block_editor() {
+		return post_type_supports( static::NAVIGATION_POST_TYPE, 'editor' );
+	}
+
+	private function create_post( $post_type ) {
+		$post            = new WP_Post( new StdClass() );
```

Comment:
```suggestion
		$post            = new WP_Post( new stdClass() );
```
1. While class names are case-insensitive, for consistency and ease of searching, we should use the class names as they are named in the PHP manual.
2. Question: Should this be a `WP_Post`, or should it use the `$this->factory()->post->create()` method? I'm not sure what the answer is here.

Reply:
1. My bad. Fixed in https://github.com/WordPress/wordpress-develop/pull/3005/commits/add6a919ddb97162e906d4821b269c9e0b856dae
2. Yes, I'm aware of `$this->factory()->post->create()`. In this particular case posts don't have to be saved to the database (that was the main reason not to use `WP_UnitTest_Factory_For_Post::create`), but I don't mind to use factory as long as tests pass. Refactored in 9d26fc76c5277641c05185e54adffcf7e4b1c2b5